cmd/suseconnect/suseconnect.go

```diff
@@ -390,3 +401,37 @@ func fileExists(path string) bool {
 func isSumaManaged() bool {
 	return fileExists("/etc/sysconfig/rhn/systemid")
 }
+
+func processToken(token string) (string, error) {
+	var reader io.Reader
+
+	if strings.HasPrefix(token, "@") {
+		tokenFilePath := strings.TrimPrefix(token, "@")
+		file, err := os.Open(tokenFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to open token file '%s': %w", tokenFilePath, err)
+		}
+		defer file.Close()
+		reader = file
+	} else if token == "-" {
+		reader = os.Stdin
+	} else {
+		return token, nil
+	}
+
+	return readTokenFromReader(reader)
```

Comment:
This way this return and the `reader` declaration can be removed.

Reply:
removed the reader